x/gov/handler.go

```diff
@@ -159,8 +162,22 @@ func (h VoteHandler) validate(ctx weave.Context, db weave.KVStore, tx weave.Tx)
 }
 
 type TallyHandler struct {
-	auth   x.Authenticator
-	bucket *ProposalBucket
+	auth        x.Authenticator
+	propBucket  *ProposalBucket
+	elecBucket  *ElectorateBucket
+	typeHandler map[Proposal_Type]TallyResultExecutor
+}
+
+func NewTallyHandler(auth x.Authenticator, bucket *ProposalBucket, electBucket *ElectorateBucket) *TallyHandler {
+	return &TallyHandler{
+		auth:       auth,
+		propBucket: bucket,
+		elecBucket: electBucket,
+		typeHandler: map[Proposal_Type]TallyResultExecutor{
```

Comment:
And this could theoretically be passed into the x/gov module from the app, allowing many types refering to different modules.

Reply:
Yes, as a global registry or we do a router like approach based on `Proposal_Type`.